src/ctnetlink/nlas/flow/nla.rs

```diff
@@ -0,0 +1,120 @@
+// SPDX-License-Identifier: MIT
+
+use byteorder::{BigEndian, ByteOrder};
+use netlink_packet_utils::{
+    nla::{DefaultNla, Nla, NlaBuffer, NLA_F_NESTED},
+    Emitable, Parseable,
+};
+
+use crate::constants::{
+    CTA_ID, CTA_MARK, CTA_PROTOINFO, CTA_STATUS, CTA_TIMEOUT, CTA_TUPLE_ORIG,
+    CTA_TUPLE_REPLY, CTA_USE,
+};
+
+use super::{
+    ip_tuple::TupleNla, protocol_info::ProtocolInfo, status::ConnectionStatus,
+};
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub enum FlowNla {
+    Orig(Vec<TupleNla>),
+    Reply(Vec<TupleNla>),
+    Status(ConnectionStatus),
+    ProtocolInfo(ProtocolInfo),
+    Timeout(u32),
+    Mark(u32),
+    Use(u32),
+    Id(u32),
+    Other(DefaultNla),
+}
+
+impl Nla for FlowNla {
+    fn value_len(&self) -> usize {
+        match self {
+            FlowNla::Orig(attrs) => {
+                attrs.iter().fold(0, |l, attr| l + attr.buffer_len())
+            }
+            FlowNla::Reply(attrs) => {
+                attrs.iter().fold(0, |l, attr| l + attr.buffer_len())
+            }
+            FlowNla::Status(attr) => attr.value_len(),
+            FlowNla::ProtocolInfo(attr) => attr.value_len(),
+            FlowNla::Timeout(_) => 4,
+            FlowNla::Mark(_) => 4,
+            FlowNla::Use(_) => 4,
+            FlowNla::Id(_) => 4,
+            FlowNla::Other(attr) => attr.value_len(),
+        }
+    }
+
+    fn kind(&self) -> u16 {
+        match self {
+            FlowNla::Orig(_) => CTA_TUPLE_ORIG | NLA_F_NESTED,
```

Comment:
`NLA_F_NESTED` should be handled automatically by upper layer crate like `netlink-packet-core`, so you do not need to set it here.